lib/Gocdb_Services/NotificationService.php

```diff
@@ -97,33 +73,14 @@ public function roleRequest($entity) {
         } else {
             // If the entity has valid users who can approve the role then send the email notification.
 
-            // Remove duplicate emails from array
-            $emails = array_unique ( $emails );
-
-            // Get the PortalURL to create an accurate link to the role approval view
-            $localInfoLocation = __DIR__ . "/../../config/local_info.xml";
-            $localInfoXML = simplexml_load_file ( $localInfoLocation );
-            $webPortalURL = $localInfoXML->local_info->web_portal_url;
-
-            // Email content
-            $headers = "From: no-reply@goc.egi.eu";
-            $subject = "GocDB: A Role request requires attention";
+            // Remove duplicate user ids from array
+            $authorisingUserIds = array_unique ( $authorisingUserIds );
 
-            $body = "Dear GOCDB User,\n\n" . "A user has requested a role that requires attention.\n\n" .
-                    "You can approve or deny this request here:\n\n" . $webPortalURL . "/index.php?Page_Type=Role_Requests\n\n" .
-                    "Note: This role may already have been approved or denied by another GocDB User";
-
-            $sendMail = TRUE;
             // Send email to all users who can approve this role request
-            if ($emails != null) {
-                foreach ( $emails as $email ) {
-                    if($sendMail){
-                       mail($email, $subject, $body, $headers);
-                    } else {
-                       echo "Email: " . $email . "<br>";
-                       echo "Subject: " . $subject . "<br>";
-                       echo "Body: " . $body . "<br>";
-                    }
+            if (!empty($authorisingUserIds)) {
+                foreach ( $authorisingUserIds as $userId ) {
+                    $approvingUser = \Factory::getUserService()->getUser($userId);
+                    $this->sendEmail($roleRequested, $requestingUser, $entity->getName(), $approvingUser);
                 }
             }
```

Comment:
Haven't we already checked `if (count($authorisingUserIds) == 0) {` above and array_unique won't empty it?
```suggestion
           foreach ( $authorisingUserIds as $userId ) {
                $approvingUser = \Factory::getUserService()->getUser($userId);
                $this->sendEmail($roleRequested, $requestingUser, $entity->getName(), $approvingUser);
            }
```

Reply:
Good point!